dask/dask.yaml

```diff
@@ -15,7 +15,7 @@ dataframe:
     metadata-task-size-local: 512  # Number of files per local metadata-processing task
     metadata-task-size-remote: 1  # Number of files per remote metadata-processing task
   convert-string: null  # Whether to convert string-like data to pyarrow strings
-  query-planning: false  # Whether to use dask-expr
+  query-planning: null  # Whether to use dask-expr
```

Comment:
Perfect - Thank you for doing this. The RAPIDS transition to dask-expr will be much smoother if we can check whether or not the user has configured the option themselves.

Reply:
I don't fully understand how this is helpful for downstream libraries. Because with the above that means I can't simply do

```
from dask import config
QUERY_PLANNING_ON = config.get("dataframe.query-planning", False)
```

to know if dask itself will be using the dask-expr based version (need to add something like `if QUERY_PLANNING_ON is None: QUERY_PLANNING_ON = True`)

> if we can check whether or not the user has configured the option themselves.

Do you mean that you will turn off dask-expr when not explicitly enabled by the user in cudf? 
But at that point `dask.dataframe` already imported things assuming it is using dask-expr

Reply:
Good question @jorisvandenbossche - Sorry, it's actually not very useful for most down-stream libraries. It's also not something we are *planning* to use.

Having a null default does make it possible for rapids to intercept a `dask.dataframe` import and override the default config. However, this is not something we are planning to do unless there is no other option (e.g. a critical bug is found in `2024.1.1`).

I do realize that it is a bit annoying to do:

```
from dask import config
QUERY_PLANNING_ON = config.get("dataframe.query-planning", None) is not False
```

but for a library offering a `dask.dataframe.DataFrame` API wrapper (e.g. dask-cudf), it's nice for it to be possible know if the user has explicitly opted in/out of query planning.

Reply:
It’s also not something that I am super happy about but it’s better than the alternative as long as we support pandas < 2